### PR TITLE
Image upload: display mode

### PR DESCRIPTION
**Pull request by ramongduraes**
Adding a display mode to the Image Upload component and updating the documentation. 

Service providers may now:
- Simply keep the input image on screen when the service is complete (refer to `disableResetButton` prop);
- Provide the output image to the component and be able to display the input and output images, compare them through an interactive tab and allow the user to download the output image.

### CODE REVIEW

File: src/components/service/standardComponents/SNETImageUpload.js [hunk 1]
```diff
@@ -127,11 +129,121 @@ export default class SNETImageUpload extends React.Component {
...
+        this.inputImageErrorMessage = "Input image could not be rendered.";
+        this.outputImageErrorMessage = "Output image could not be rendered.";
+
+        // Refs
+        this.imageDiv = React.createRef();
+        this.inputImage = React.createRef();
+        this.outputImage = React.createRef();
 
         // Function binding
+        this.componentWillReceiveProps = this.componentWillReceiveProps.bind(this);
```
Comment (raamb) [inline src/components/service/standardComponents/SNETImageUpload.js]: There is no need to bind componentWillReceiveProps

**Comment by ferrouswheel** [comment]
Do you have any screenshots?

Since I do this alot, I wonder if one of the patterns I use could be included...

I overlay a canvas on an img element, and then annotate the image by drawing to the canvas based on the service results.

(Not suggesting this should be part of this pull request! Just an idea :-) )

**Comment by ramongduraes** [comment]
Yes! So, I've been using the component in 2 ways:

1. You can send `true` to `disableResetButton` once the service is complete to display the input image to the user (it simply removes any reset options).  The result is this:
screenshot from 2019-02-25 07-45-44
2. You can also send the base64-encoded `outputImage` to the component. Once it is truthy, the component will switch to the display mode and render 3 tabs (input, output, comparison):
screenshot from 2019-02-25 07-45-54
screenshot from 2019-02-25 07-45-25

Right now you'd have to draw the boxes on top of the image element and send the resulting base64 encoded image to the component to use it. Sounds like it's more work... We should probably consider adding a way to draw on top of the input/output canvases!

**Comment by ferrouswheel** [comment]
Nice! The comparison slider sure looks snazzy :-)

**Pull request merged**